pint/testsuite/test_issues.py

```diff
@@ -886,6 +886,15 @@ def test_issue_1300(self):
         m = module_registry.Measurement(1, 0.1, "meter")
         assert m.default_format == "~P"
 
+    @helpers.requires_numpy()
+    def test_issue1674(self, module_registry):
+        Q_ = module_registry.Quantity
+        arr_of_q = np.array([Q_(2, "m"), Q_(4, "m")], dtype="object")
+        q_arr = Q_(np.array([1, 2]), "m")
+
+        helpers.assert_quantity_almost_equal(arr_of_q * q_arr, Q_([2, 8], "m^2"))
+        helpers.assert_quantity_almost_equal(arr_of_q / q_arr, Q_([2, 2], ""))
```

Comment:
This doesn't look quite right to me: You start from an `dtype=object` array of quantity scalars, where each element could potentially be of a different unit. The result is now a "normal" quantity array, where all elements share a single unit.

Please add a test case for `arr_of_q = np.array([Q_(2, "m"), Q_(4, "s")], dtype="object")`.

Obviously, this will not work with a quantity array, and the expected result of `arr_of_q * q_arr` would be `np.array([Q_(2, "m^2"), Q_(4, "m s")], dtype="object")`. For consistency, I would have expected to receive a `dtype=object` array in all cases.